components/SideBar.jsx

```diff
@@ -1,3 +1,79 @@
+import { useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faPersonBiking,
+  faPersonWalking,
+  faPersonSwimming,
+  faPersonHiking,
+  faPersonRunning,
+  faSquarePlus,
+  faCalculator,
+} from "@fortawesome/free-solid-svg-icons";
+
 export function SideBar() {
-  return <p>Side Bar</p>;
+  const [activeMenu, setActiveMenu] = useState("");
+
+  const handleMenuClick = (menuName) => {
+    setActiveMenu(menuName);
```

Comment:
ให้เปลี่ยนไปใช้ setActiveMenu แบบนี้แทนดีมั้ย คือถ้ามันกดซ้ำที่เดิมอีกรอบ มันจะเคลียร์ active ออก ประมาณว่า กดซ้ำอีกรอบก็คือไม่เลือกแล้ว อะไรอย่างงี้ อย่างงี้ 😉😉
```suggestion
    setActiveMenu((currentActiveMenu) => menuName === currentActiveMenu ? "" : menuName);
```